app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -69,6 +94,7 @@ def create_vm(args = {})
       message = JSON.parse(e.response.body)['badRequest']['message'] rescue (e.to_s)
       logger.warn "failed to create vm: #{message}"
       destroy_vm vm.id if vm
+      delete_volume(@boot_vol_id) if args[:boot_from_volume] == "true"
```

Comment:
`volume_client.volumes.delete(@boot_vol_id)` should be fine, no need to extract 1 clear line to a method in my opinion